mlptrain/potentials/mace/mace.py

```diff
@@ -0,0 +1,556 @@
+import numpy as np
+from time import time
+from copy import deepcopy
+from mlptrain.box import Box
+from mlptrain.log import logger
+from mlptrain.config import Config
+from mlptrain.potentials._base import MLPotential
+import ast
+import logging
+import os
+from typing import Optional
+import torch.nn.functional
+from e3nn import o3
+from torch.optim.swa_utils import SWALR, AveragedModel
+from torch_ema import ExponentialMovingAverage
+import mace
+from mace import data, modules, tools
+from mace.tools import torch_geometric,  evaluate
+from mlptrain.config import Config
+from ase.data import chemical_symbols
+import dataclasses
+from prettytable import PrettyTable
+import torch
+
+
+@dataclasses.dataclass
+class SubsetCollection:
+    train: data.Configurations
+    valid: data.Configurations
+
+class MACE(MLPotential):
+
+    def _train_file(self,
```

Comment:
This is a huge function, probably worth splitting it up into smaller functions as it is difficult to follow what is going on